spring-boot-project/spring-boot-autoconfigure/build.gradle

```diff
@@ -58,6 +58,7 @@ dependencies {
 	optional("org.apache.commons:commons-dbcp2")
 	optional("org.apache.httpcomponents.client5:httpclient5")
 	optional("org.apache.kafka:kafka-streams")
+	optional("org.apache.qpid:proton-j")
```

Comment:
I'm strongly to spot why this is needed so I tried removing it and everything compiled cleanly. If it isn't needed we can get rid of the dependency management too.

Reply:
We could move it to `testImplementation` but the stream-client is practically unusable without it; Spring Initializr would have to add it anyway - I don't know why it is not a transitive dependency of the client. I can start a discussion with them.

```
...
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.rabbitmq.stream.Environment]: Factory method 'rabbitStreamEnvironment' threw exception; nested exception is java.lang.NoClassDefFoundError: org/apache/qpid/proton/amqp/messaging/Section
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
	... 128 more
Caused by: java.lang.NoClassDefFoundError: org/apache/qpid/proton/amqp/messaging/Section
```

Reply:
@acogoluegnes Agreed: https://github.com/rabbitmq/rabbitmq-stream-java-client/issues/18